docusaurus/docs/iOS/swiftui/swiftui-cookbook/custom-channel-list.md

```diff
@@ -0,0 +1,237 @@
+    didChangeChannels changes: [ListChange<ChatChannel>]
+) {
+    channels = controller.channels
+}
+```
+
+In order to support pagination, we will use the `ChatChannelListController`'s method `loadNextChannels`. We can call this method when the channel list items appear on the screen.
+
+```swift
+public func checkForChannels(index: Int) {
+    if index < (controller?.channels.count ?? 0) - 15 {
+        return
+    }
+
+    if !loadingNextChannels {
+        loadingNextChannels = true
+        controller?.loadNextChannels(limit: 30) { [weak self] _ in
+            guard let self = self else { return }
+            self.loadingNextChannels = false
+        }
+    }
+}
+```
+
+Those are the most important bits for fetching and listening to the channels data. For reference, here's the full implementation of the view model.
+
+```swift
+import StreamChat
+import StreamChatSwiftUI
+import SwiftUI
+
+class CustomChannelListViewModel: ObservableObject, ChatChannelListControllerDelegate {
+    
+    @Injected(\.chatClient) var chatClient
+    
+    @Published public var channels = LazyCachedMapCollection<ChatChannel>()
+    
+    public private(set) var loadingNextChannels: Bool = false
+    
+    private var controller: ChatChannelListController?
+    private var timer: Timer?
+    
+    init() {
+        setupChannelListController()
+    }
+    
+    public func controller(
+        _ controller: ChatChannelListController,
+        didChangeChannels changes: [ListChange<ChatChannel>]
+    ) {
+        channels = controller.channels
+    }
+    
+    public func checkForChannels(index: Int) {
+        if index < (controller?.channels.count ?? 0) - 15 {
+            return
+        }
+
+        if !loadingNextChannels {
+            loadingNextChannels = true
+            controller?.loadNextChannels(limit: 30) { [weak self] _ in
+                guard let self = self else { return }
+                self.loadingNextChannels = false
+            }
+        }
+    }
+    
+    func imageURL(for channel: ChatChannel) -> URL? {
+        channel.lastActiveMembers.first(where: { member in
+            member.id != chatClient.currentUserId
+        })?.imageURL
+    }
+    
+    //MARK: - private
+    
+    private func updateChannels() {
+        channels = controller?.channels ?? LazyCachedMapCollection<ChatChannel>()
+    }
+    
+    private func setupChannelListController() {
+        guard let currentUserId = chatClient.currentUserId else {
+            observeClientIdChange()
+            return
+        }
+        controller = chatClient.channelListController(
+            query: .init(filter: .containMembers(userIds: [currentUserId]))
+        )
+        
+        controller?.delegate = self
+
+        updateChannels()
+
+        controller?.synchronize { [weak self] error in
+            guard let self = self else { return }
+            if error != nil {
+                // handle error
+                print("handle error here")
```

Comment:
Do we need a comment and a print stating the same?

Reply:
removed the comment